### PR TITLE
Cleanup introduction of links element

### LINKED ISSUE #1110: Improve definitions in introduction
https://manual.nexusformat.org/introduction.html
Explanation of “Links” uses constructs (NXdata, NXmonitor…) which are undefined at this point of the documentation. The definitions in the design chapter (https://manual.nexusformat.org/design.html#nexus-objects-and-terms) seem clearer than these definitions in the introduction.

### PR DESCRIPTION
**Pull request by prjemian**
- close #1110

### CODE REVIEW

**Comment by prjemian** [comment]
@BalazsBago - Please comment.

File: manual/source/introduction.rst [hunk 1]
```diff
@@ -146,14 +146,11 @@ links.
...
-   NXdata (base class); plotting
-   NXmonitor (base class); plotting
-   NXdetector (base class); plotting
    
 :ref:`Design-Links`
-    Links are used to reference the plottable data from ``NXdata``
-    when the data is provided in other groups
-    such as ``NXmonitor`` or ``NXdetector``.
+    Links are pointers to existing data somewhere else in the hierarchy. They
+    are used to have the same information be represented in different places
```
Comment (PeterC-DLS) [inline manual/source/introduction.rst:152]: A bit convoluted. Maybe better as "are used to represent the same information in different places".

Comment (prjemian) [reply manual/source/introduction.rst]: @PeterC-DLS Agree?

**Review by PeterC-DLS — APPROVED** [review]
LGTM

**Pull request merged**